Create _make_alias to preserve the name of a function when creating an alias

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88117
* #88116
* #88115
* __->__ #88114

Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

cc @ezyang @mruberry @ngimel @Lezcano @fdrocha @peterbell10

lezcano committed:
Create _make_alias to preserve the name of a function when creating an alias

Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

[ghstack-poisoned]

mruberry commented on torch/_refs/__init__.py:
```diff
@@ -413,6 +413,17 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
     return inner
 
 
+def _make_alias(fn, name=None):
```
Comment describing this function and how it's intended to be used

mruberry commented on torch/_refs/__init__.py:
```diff
@@ -413,6 +413,17 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
 
 
+def _make_alias(fn, name=None):
+    if name is None:
```
Comment describing this parsing

mruberry commented on torch/_refs/__init__.py:
```diff
@@ -413,6 +413,17 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
     return inner
 
 
+def _make_alias(fn, name=None):
```
Is `name=None` actually an interesting case to support? What's an example where a developer would want to do that instead of explicitly specifying a name?

lezcano replied:
The parsing was wrong, and you were right. I think it's best to remove the parsing altogether.

lezcano committed:
Update on "Create _make_alias to preserve the name of a function when creating an alias"


Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

cc ezyang mruberry ngimel Lezcano fdrocha

[ghstack-poisoned]

lezcano committed:
Update on "Create _make_alias to preserve the name of a function when creating an alias"


Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

cc ezyang mruberry ngimel Lezcano fdrocha

[ghstack-poisoned]

lezcano committed:
Update on "Create _make_alias to preserve the name of a function when creating an alias"


Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

cc ezyang mruberry ngimel Lezcano fdrocha

[ghstack-poisoned]

lezcano committed:
Update on "Create _make_alias to preserve the name of a function when creating an alias"


Before, we would inherit the name of the aliased function, which was
very confusing, and disallowed some homogeneous treatment of references,
as we do later in this stack

cc ezyang mruberry ngimel Lezcano fdrocha

[ghstack-poisoned]

mruberry reviewed (approved):
Cool

lezcano commented:
@pytorchbot merge
